resolver_mac.c

```diff
@@ -148,10 +148,16 @@ bool proxy_resolver_mac_get_proxies_for_url(void *ctx, const char *url) {
 
         CFRunLoopSourceRef run_loop = CFNetworkExecuteProxyAutoConfigurationURL(
             url_ref, target_url_ref, proxy_resolver_mac_auto_config_result_callback, &context);
+        if (!run_loop) {
+            proxy_resolver->error = ELOOP;
+            LOG_ERROR("Failed to execute pac url (%" PRId32 ")\n", proxy_resolver->error);
+            goto mac_done;
+        }
 
         CFRunLoopAddSource(CFRunLoopGetCurrent(), run_loop, PROXY_RESOLVER_RUN_LOOP);
         CFRunLoopRunInMode(PROXY_RESOLVER_RUN_LOOP, PROXY_RESOLVER_TIMEOUT_SEC, false);
         CFRunLoopRemoveSource(CFRunLoopGetCurrent(), run_loop, PROXY_RESOLVER_RUN_LOOP);
+        CFRelease(run_loop);
```

Comment:
As for [WebKit](https://github.com/apple-oss-distributions/WebCore/blob/d8528fda3e9828ad9a6d3d7d3bd899a671abe7b1/platform/network/cf/ProxyServerCFNet.cpp#L84), `CFRelease(run_loop)` must be called after `CFRunLoopSourceInvalidate(run_loop)`.

Reply:
Ok, I have added `CFRunLoopSourceInvalidate`.